Components/Transforms/AdvancedLimitedEulerTransform/itkAdvancedLimitedEuler3DTransform.hxx

```diff
@@ -0,0 +1,622 @@
+    RotationX[1][0] = 0; RotationX[1][1] = cx; RotationX[1][2] = -sx;
+    RotationX[2][0] = 0; RotationX[2][1] = sx; RotationX[2][2] = cx;
+
+    Matrix< TScalarType, 3, 3 > RotationY;
+    RotationY[0][0] = cy; RotationY[0][1] = 0; RotationY[0][2] = sy;
+    RotationY[1][0] = 0; RotationY[1][1] = 1; RotationY[1][2] = 0;
+    RotationY[2][0] = -sy; RotationY[2][1] = 0; RotationY[2][2] = cy;
+
+    Matrix< TScalarType, 3, 3 > RotationZ;
+    RotationZ[0][0] = cz; RotationZ[0][1] = -sz; RotationZ[0][2] = 0;
+    RotationZ[1][0] = sz; RotationZ[1][1] = cz; RotationZ[1][2] = 0;
+    RotationZ[2][0] = 0; RotationZ[2][1] = 0; RotationZ[2][2] = 1;
+
+    // derivative matrices
+    Matrix< TScalarType, 3, 3 > RotationXd;
+    RotationXd[0][0] = 0; RotationXd[0][1] = 0; RotationXd[0][2] = 0;
+    RotationXd[1][0] = 0; RotationXd[1][1] = cxd; RotationXd[1][2] = -sxd;
+    RotationXd[2][0] = 0; RotationXd[2][1] = sxd; RotationXd[2][2] = cxd;
+
+    Matrix< TScalarType, 3, 3 > RotationYd;
+    RotationYd[0][0] = cyd; RotationYd[0][1] = 0; RotationYd[0][2] = syd;
+    RotationYd[1][0] = 0; RotationYd[1][1] = 0; RotationYd[1][2] = 0;
+    RotationYd[2][0] = -syd; RotationYd[2][1] = 0; RotationYd[2][2] = cyd;
+
+    Matrix< TScalarType, 3, 3 > RotationZd;
+    RotationZd[0][0] = czd; RotationZd[0][1] = -szd; RotationZd[0][2] = 0;
+    RotationZd[1][0] = szd; RotationZd[1][1] = czd; RotationZd[1][2] = 0;
+    RotationZd[2][0] = 0; RotationZd[2][1] = 0; RotationZd[2][2] = 0;
+
+    // Aply the rotation first around Y then X then Z
+    if (m_ComputeZYX)
+    {
+      jsj[0] = RotationZ * RotationY * RotationXd;
+      jsj[1] = RotationZ * RotationYd * RotationX;
+      jsj[2] = RotationZd * RotationY * RotationX;
+
+    }
+    else
+    {
+      // Like VTK transformation order
+      jsj[0] = RotationZ * RotationXd * RotationY;
+      jsj[1] = RotationZ * RotationX * RotationYd;
+      jsj[2] = RotationZd * RotationX * RotationY;
+    }
+
+    // Translation parameters:
+    for (unsigned int par = 3; par < ParametersDimension; ++par)
+    {
+      jsj[par].Fill(0.0);
+    }
+  }
+
+
+  // Print self
+  template< class TScalarType >
+  void
+    AdvancedLimitedEuler3DTransform< TScalarType >::PrintSelf(std::ostream& os, Indent indent) const
+  {
+    Superclass::PrintSelf(os, indent);
+
+    os << indent << "Euler's angles: AngleX=" << m_AngleX
+      << " AngleY=" << m_AngleY
+      << " AngleZ=" << m_AngleZ
+      << std::endl;
+    os << indent << "m_ComputeZYX = " << m_ComputeZYX << std::endl;
+  }
+
+
+  template <typename TParametersValueType>
+  void
+    AdvancedLimitedEuler3DTransform<TParametersValueType>::InitLimitParameters()
+  {
+    m_ScalesEstimation = false;
+    m_SharpnessOfLimits = 50.0f;
+
+    m_UpperLimits.SetSize(ParametersDimension);
+    m_LowerLimits.SetSize(ParametersDimension);
+    for (unsigned int i = 0; i < ParametersDimension / 2; i++)
+    {
+      m_UpperLimits[i] = 3.14f;
+      m_LowerLimits[i] = -3.14f;
```

Comment:
If you mean &#x3C0;, please use `vnl_math::pi` (and then do `#include <vnl/vnl_math.h>`.) As an alternative, `M_PI` (with ``#include <cmath>`) is also fine to me.